RTech.CoffeeMachine.Api.Tests/BrewCoffee/Controllers/CoffeeMachineControllerTests.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System.Net;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.Caching.Memory;
+using Moq;
+using RTech.CoffeeMachine.Api.BrewCoffee.Controllers;
+using RTech.CoffeeMachine.Api.BrewCoffee.Models;
+using RTech.CoffeeMachine.Api.BrewCoffee.Util;
+using Shouldly;
+
+namespace RTech.CoffeeMachine.Api.Tests.BrewCoffee.Controllers;
+
+public class CoffeeMachineControllerTests : IClassFixture<WebApplicationFactory<Program>>
+{
+    private readonly WebApplicationFactory<Program> _factory;
+
+    public CoffeeMachineControllerTests(WebApplicationFactory<Program> factory)
+    {
+        _factory = factory;
+    }
+
+    [Fact]
+    public async Task GetBrewStatus_WhenUnrecognizedBrewStatus_ShouldReturn500()
+    {
+        var brewStatusProvider = new Mock<IBrewStatusProvider>();
+        brewStatusProvider.Setup(p => p.GetBrewStatus())
+            .Returns(BrewStatus.None);
+
+        var response = await GetEndpointResponse(services =>
+        {
+            services.AddSingleton<IBrewStatusProvider>(brewStatusProvider.Object);
+        });
+
+        response.StatusCode.ShouldBe(HttpStatusCode.InternalServerError);
+        var content = await response.Content.ReadAsStringAsync();
+        content.ShouldBe(string.Empty);
+    }
+
+    [Fact]
+    public async Task GetBrewStatus_WhenException_ShouldReturn500()
+    {
+        var dateTimeProvider = new Mock<IDateTimeProvider>();
+        dateTimeProvider.Setup(p => p.GetLocalNow())
+            .Throws<InvalidOperationException>();
+
+        var response = await GetEndpointResponse(services =>
+        {
+            services.AddSingleton<IDateTimeProvider>(dateTimeProvider.Object);
+        });
+
+        response.StatusCode.ShouldBe(HttpStatusCode.InternalServerError);
+        var content = await response.Content.ReadAsStringAsync();
+        content.ShouldBe(string.Empty);
+    }
+
+    [Fact]
+    public async Task GetBrewStatus_WhenBrewStatusUnavailable_ShouldReturnEmpty503()
+    {
+        var cache = new MemoryCache(new MemoryCacheOptions());
+        long numRequestsSoFar = 4;
+        cache.Set(CoffeeMachineController.CACHE_KEY_NUM_REQUESTS, numRequestsSoFar);
+        var response = await GetEndpointResponse(services =>
+        {
+            services.AddSingleton<IMemoryCache>(cache);
+        });
+
+        response.StatusCode.ShouldBe(HttpStatusCode.ServiceUnavailable);
+        var content = await response.Content.ReadAsStringAsync();
+        content.ShouldBe(string.Empty);
+    }
+
+    [Theory]
+    [InlineData(0, StatusCodes.Status200OK)]
+    [InlineData(1, StatusCodes.Status200OK)]
+    [InlineData(2, StatusCodes.Status200OK)]
+    [InlineData(3, StatusCodes.Status200OK)]
+    [InlineData(4, StatusCodes.Status503ServiceUnavailable)]
+    [InlineData(5, StatusCodes.Status200OK)]
```

Comment:
A bit of test coverage overlap here, but the overlap is minimal & I consider it worth it.
In this scenario, I usually try to encourage the PR reviewer to voice their preference (if they have one).